reV/offshore/offshore.py

```diff
@@ -441,49 +461,71 @@ def run_nrwal(self):
+                logger.warning(msg)
+                warn(msg)
+            elif np.isnan(arr).any():
                 mask = np.isnan(arr)
                 nan_meta = self.meta_out_offshore[mask]
                 nan_gids = nan_meta['gid'].values
                 msg = ('NaN values ({} out of {}) persist in offshore '
                        'output "{}"!'
                        .format(np.isnan(arr).sum(), len(arr), name))
-                logger.error(msg)
-                logger.error('This is the offshore meta that is causing NaN '
-                             'outputs: {}'.format(nan_meta))
-                logger.error('These are the resource gids causing NaN '
-                             'outputs: {}'.format(nan_gids))
-                raise ValueError(msg)
+                logger.warning(msg)
+                logger.warning('This is the offshore meta that is causing NaN '
+                               'outputs: {}'.format(nan_meta))
+                logger.warning('These are the resource gids causing NaN '
+                               'outputs: {}'.format(nan_gids))
+                warn(msg)
 
     def write_to_gen_fpath(self):
         """Save offshore outputs to input generation fpath file. This will
         overwrite data!"""
 
         loss_mult = 1 - self._out[self._loss_key]
+        loss_mult = np.where(np.isnan(loss_mult), 1, loss_mult)
 
         with Outputs(self._gen_fpath, 'a') as f:
             meta_attrs = f.get_attrs('meta')
             del f._h5['meta']
             f._set_meta('meta', self.meta_out, attrs=meta_attrs)
 
-            lcoe = f['lcoe_fcr']
-            lcoe[self._offshore_mask] = self._out[self._lcoe_key]
-            f['lcoe_fcr'] = lcoe
-
-            cf_mean = f['cf_mean']
-            cf_mean[self._offshore_mask] *= loss_mult
-            f['cf_mean'] = cf_mean
-
-            if 'cf_profile' in f.dsets:
+            if 'lcoe_fcr' in f:
+                lcoe = f['lcoe_fcr']
+                lcoe[self._offshore_mask] = self._out[self._lcoe_key]
+                f['lcoe_fcr'] = lcoe
+            elif 'lcoe_fcr' not in f and self._run_all:
+                f._add_dset('lcoe_fcr', self._out[self._lcoe_key], np.float32,
+                            attrs={'units': 'dol/MWh', 'scale_factor': 1})
+            elif 'lcoe_fcr' not in f and not self._run_all:
+                lcoe = np.zeros(len(self.meta_out), dtype=np.float32)
+                lcoe[self._offshore_mask] = self._out[self._lcoe_key]
+                f._add_dset('lcoe_fcr', lcoe, np.float32,
+                            attrs={'units': 'dol/MWh', 'scale_factor': 1})
+
+            f._add_dset('cf_mean_raw', self._cf_mean, f.dtypes['cf_mean'],
+                        attrs=f.attrs['cf_mean'])
+            self._cf_mean[self._offshore_mask] *= loss_mult
+            f['cf_mean'] = self._cf_mean
+
+            profiles = None
+            if 'cf_profile_raw' in f.dsets:
```

Comment:
what is the difference between cf_profile_raw and cf_profile

Reply:
the _raw datsets are CF before NRWAL losses are applied. We save the raw datasets so we can rerun NRWAL and not doubly count losses.

Reply:
Sweet, that was my guess!